apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex

```diff
@@ -63,6 +63,10 @@ defmodule Lexical.RemoteControl.CodeIntelligence.Entity do
     resolve_alias(chars, node_range, analysis, position)
   end
 
+  defp resolve({:local_or_var, 'defstruct'}, node_range, _analysis, _position) do
```

Comment:
Why does this need to be special-cased? Shouldn't this be resolved by looking at imports?

For instance, while ill-advised, one could shadow `defstruct` and use their own wrapper, e.g.

```elixir
defmodule MyModule do
  import Kernel, except: [defstruct: 1]
  import MyDefstruct
  
  # from MyDefstruct
  defstruct x: 1, y: 2
end
```

Reply:
It's a little unclear to me how entity should handle this generically though. There's a local or var case already, but that seems to want to return a 0 arity call.

Reply:
worse, i cannot find a way to reliably determine the arity of the function in question if it doesn't have parens.

Reply:
ok, i took another stab at it.